client/bin/multi-client.js

```diff
@@ -474,24 +480,29 @@ class State {
 		while (runQueue.length > 0) {
 			const job = runQueue.pop();
 			// If matching workers for the job are available then allot them a job
-			if (job.matchingDevices !== null) { // specifically for an application since worker URL's are specific are automatically allocated
-				for (var device of job.matchingDevices) {
-					// check if device is idle & public URL is reachable
-					let deviceUrl = await balenaCloud.resolveDeviceUrl(device)
-					try{
-						let status = await rp.get(`${deviceUrl}/state`);
-						if (status === "IDLE") {
-							// make sure that the worker being targetted isn't already about to be used by another child process
-							if(!busyWorkers.includes(deviceUrl)){
-								// Create single client and break from loop to job the job 👍
-								job.workers = deviceUrl
+			for (var device of job.matchingDevices) {
+				// check if device is idle & public URL is reachable
+				let deviceUrl = ''
+				if(!job.array){
+					deviceUrl = await balenaCloud.resolveDeviceUrl(device)
+				} else {
+					deviceUrl = device;
+				}
+				try{
+					let status = await rp.get(`${deviceUrl}/state`);
+					if (status === "IDLE") {
+						// make sure that the worker being targetted isn't already about to be used by another child process
+						if(!busyWorkers.includes(deviceUrl)){
+							// Create single client and break from loop to job the job 👍
+							job.workers = deviceUrl
+							if(!job.array){
 								job.workerPrefix = device.fileNamePrefix()
-								break
 							}
+							break
 						}
-					} catch(err) {
-						state.info(`Couldn't retrieve ${device.tags.DUT} worker's state. Querying ${deviceUrl} and received ${err.name}: ${err.statusCode}`)
 					}
+				} catch(err) {
+					state.info(`Couldn't retrieve ${device.tags.DUT} worker's state. Querying ${deviceUrl} and received ${err.name}: ${err.statusCode}`)
```

Comment:
I get
```
ERROR ENCOUNTERED: Cannot read property 'DUT' of undefined. 
```
when running against the local qemu worker

Reply:
```suggestion
					state.info(`Couldn't retrieve ${device.tags ? device.tags.DUT : device} worker's state. Querying ${deviceUrl} and received ${err.name}: ${err.statusCode}`)
```

gives me

```
INFO: Couldn't retrieve http://localhost/ worker's state. Querying http://localhost/ and received StatusCodeError: 404
```

Reply:
I can reach the core using curl though, so not sure what the problem is:
```
$ curl -L http://localhost/state ; echo
IDLE
```

Reply:
ah ok, the uri from `config.js` is just taken unmodified... I had `http://localhost/` which becomes `http://localhost//state` which gives a 404

Reply:
We could easily fix that to remove the extra slash?